### PR TITLE
Add note about list indentation to MD029

### PR DESCRIPTION
**Pull request by Ace-Cassidy**

### CODE REVIEW

**Comment by DavidAnson** [comment]
Could you please clarify/explain the intent of this change? I’ve included the example from the change along with how I would write that same thing in the sample I link to below. Neither reports a violation, but I may have misunderstood. Thanks for the help!

https://dlaa.me/markdownlint/#%25m%23%20Issue%20265%0A%0APR%20example%3A%0A%0A1.%20item%201%0A2.%20item%202%0A%0A%20%20%20%20%5C%60%5C%60%5C%60%0A%20%20%20%20Code%20block%0A%20%20%20%20%5C%60%5C%60%5C%60%0A3.%20item%203%0A%0AThis%20is%20how%20I%20would%20expect%20to%20write%20that%3A%0A%0A1.%20item%201%0A2.%20item%202%0A%0A%20%20%20%60%60%60text%0A%20%20%20Code%20block%0A%20%20%20%60%60%60%0A%0A3.%20item%203%0A%0A4%20space%20indent%20is%20not%20needed.%0A

**Comment by Ace-Cassidy** [comment]
Thanks for your reply to my sloppy pull request.
perhaps a screenshot will clarify: https://photos.app.goo.gl/xm1yF3Mi33n8rxi5A

This rule is triggered if you don't indent properly, but going to the rules page does not make that clear.

**Comment by DavidAnson** [comment]
Thanks - as I understand the image example, the rule is correct. The code block is not indented, so it has the effect of ending the list started by the item above it. The item below it starts a new list, which the rule says should start with the number 1 and not 2. I have created an example to show what I mean - it takes advantage of auto-numbering to show when a new list is created:

https://dlaa.me/markdownlint/#%25m%23%20Issue%20265%0A%0AExample%20from%20photo%3A%0A%0A1.%20Text%0A%0A%60%60%60text%0ACode%0A%60%60%60%0A%0A1.%20Text%2C%20starts%20with%20“1”%20because%20this%20is%20a%20different%20list%20because%20the%20code%20block%20was%20not%20part%20of%20the%20previous%20item%0A%0A---%0A%0A1.%20Text%0A%0A%20%20%20%60%60%60text%0A%20%20%20Code%0A%20%20%20%60%60%60%0A%0A1.%20Text%2C%20starts%20with%20“2”%20because%20the%20code%20block%20was%20part%20of%20the%20previous%20item%0A

**Comment by Ace-Cassidy** [comment]
I see that makes sense. Perhaps on the rule information something could be added explaining this? so that when individuals such as myself make this mistake and open the link to rule it is easier to understand what is happening

**Comment by DavidAnson** [comment]
_What do you think of this:_

Note: This rule will report violations for cases like the following where an improperly-indented code block (or similar) appears between two list items and "breaks" the list in two:

~~~markdown
1. First list

```text
Code block
```

1. Second list
~~~

The fix is to indent the code block so it becomes part of the preceding list item as intended:

~~~markdown
1. First list

   ```text
   Code block
   ```

1. Still first list
~~~

**Comment by Ace-Cassidy** [comment]
That looks great! I've changed the pull request, but it looks like I did something wrong. Let me know if you want me to fix this or if you are willing to.

**Comment by DavidAnson** [comment]
GitHub is tripping you up here. The actual text to add is:

~~~~

Note: This rule will report violations for cases like the following where an improperly-indented code block (or similar) appears between two list items and "breaks" the list in two:

~~~markdown
1. First list

```text
Code block
```

1. Second list
~~~

The fix is to indent the block so it becomes part of the preceding list item as intended:

~~~markdown
1. First list

   ```text
   Code block
   ```

2. Still first list
~~~

~~~~

**Comment by DavidAnson** [comment]
The tildes above help distinguish the nested blocks. I can squash commits in the merge once you get the right formatting and the tests pass.

**Comment by Ace-Cassidy** [comment]
Looks like the checks passed this time. Thanks for working on this with me :)

**Comment by DavidAnson** [comment]
Seems good! I’ll have a second look tomorrow when I’m fresh and expect to merge the PR then. Appreciate the help!

File: doc/Rules.md [hunk 1]
```diff
@@ -992,6 +992,30 @@ This rule supports 0-prefixing ordered list items for uniform indentation:
...
+The fix is to indent the code block so it becomes part of the preceding list item as intended:
+
+~~~markdown
+1. First list
+
+   ```text
+   Code block
+   ```
+
+1. Still first list
```
Comment (DavidAnson) [inline doc/Rules.md]: Tiny request: Please change this 1 to a 2 to make it a little clearer this is the second item of the same list.

**Comment by DavidAnson** [comment]
Thanks!

**Comment by DavidAnson** [comment]
I’m not sure why GitHub isn’t showing you as the author of this commit; I accepted as a squash in the web interface as I have done in the past.

**Pull request merged**